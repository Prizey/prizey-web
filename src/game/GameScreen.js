import React from 'react'
import { List } from 'croods'

import GoBack from 'design/GoBack/GoBack'
import Caption from 'design/Caption/Caption'
import Layout from 'design/Layout/Layout'
import Roulette from './Roulette'

const InnerComponent = ({ speed, difficulty, list }) => (
  <Layout
    leftIcon={<GoBack to="/" />}
    caption={<Caption difficulty={difficulty} />}
  >
    <Roulette
      speed={speed}
      data={list}
      onSelectItem={item =>
        // eslint-disable-next-line no-alert
        window.alert(`you tapped the item ${item.title}`)
      }
    />
  </Layout>
)

const SpeedComponent = difficulty => list => (
  <List
    name="game_settings"
    path={`/game_settings`}
    render={speedList => {
      const speed = speedList[0][`${difficulty}CarouselSpeed`]

      return InnerComponent({
        difficulty,
        list,
        speed,
      })
    }}
  />
)

export default ({ difficulty = 'easy' }) => (
  <List
    name="products"
    path={`/products/${difficulty}`}
    render={SpeedComponent(difficulty)}
  />
)
