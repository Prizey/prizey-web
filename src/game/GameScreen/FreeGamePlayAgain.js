import React from 'react'
import { connect } from 'react-redux'
import { Button, Typography } from '@material-ui/core'

import UserBalance from 'design/UserBalance'
import Layout from 'design/Layout/Layout'
import ProductImage from 'design/ProductImage'
import Caption from 'design/Caption/Caption'
import placeholder from './cardi_placeholder.png'

const Screen = ({ currentUser, location, product }) => {
  const mainText = product
    ? product.title
    : `Looks like you've already played this game, Click the "Play Again" below to Download Another App and play a new game`
  const productToShow = product || {
    image: placeholder,
  }

  return (
    <Layout
      location={location}
      currentUser={currentUser}
      leftIcon={<UserBalance />}
      confirmLeave
      caption={<Caption difficulty="easy" />}
    >
      <Typography align="center" variant="h5">
        {mainText}
      </Typography>

      <ProductImage product={productToShow} />

      <a
        href="https://www.verifyspot.net/cl.php?id=c6ab41e68cc1dd56233aa14ec2ed9859"
        target="_blank"
        rel="noopener noreferrer"
        style={{ textDecoration: 'none' }}
      >
        <Button
          variant="contained"
          style={{ backgroundColor: '#76fc03' }}
          fullWidth
        >
          Play Again
        </Button>
      </a>
    </Layout>
  )
}

export const mapState = ({ basket = {} }) => ({
  product: basket.product,
})

export default connect(mapState)(Screen)
