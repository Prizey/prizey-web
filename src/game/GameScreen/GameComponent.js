import React from 'react'
import { connect } from 'react-redux'
import { Typography } from '@material-ui/core'

import { chooseProduct } from 'store/basket/actions'

import UserBalance from 'design/UserBalance'
import Caption from 'design/Caption/Caption'
import Layout from 'design/Layout/Layout'
import ProfileLink from 'design/ProfileLink/ProfileLink'
import Roulette from '../Roulette'

export default connect(
  null,
  { chooseProduct },
)(
  ({
    speed,
    difficulty,
    list,
    multiplier,
    currentUser,
    location,
    freegame,
    hideLogin,
    ...props
  }) => (
    <Layout
      leftIcon={<UserBalance />}
      caption={<Caption difficulty={difficulty} />}
      location={location}
      currentUser={currentUser}
      rightIcon={!hideLogin && <ProfileLink confirmLeave />}
      confirmLeave
    >
      <Roulette
        aria-label="roulette"
        speed={speed}
        data={list}
        multiplier={multiplier}
        onSelectItem={item => {
          props.chooseProduct(item)
          if (freegame) {
            props.navigate(`/freegame-play-again`)
          } else {
            props.navigate(`/game/${difficulty}/claim`)
          }
        }}
      />
      <br />
      <Typography align="center" variant="body2" color="textSecondary">
        Tap the screen, <br />
        win what you tap.
      </Typography>
    </Layout>
  ),
)
