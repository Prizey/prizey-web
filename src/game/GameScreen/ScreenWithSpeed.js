import React from 'react'
import SpeedComponent from '../SpeedComponent'
import GameComponent from './GameComponent'

export default (difficulty, navigate, currentUser, location) => list => (
  <SpeedComponent
    render={settings => {
      const speed = settings[`${difficulty}CarouselSpeed`]

      return (
        <GameComponent
          location={location}
          currentUser={currentUser}
          list={list}
          difficulty={difficulty}
          multiplier={settings.priceMultiplier}
          navigate={navigate}
          speed={speed}
        />
      )
    }}
  />
)
