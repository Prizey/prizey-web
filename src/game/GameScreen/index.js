import React from 'react'
import { List, Info } from 'croods'
import { Redirect } from '@reach/router'
import ScreenWithSpeed from './ScreenWithSpeed'

export default ({
  freegame,
  currentUser,
  location,
  navigate,
  difficulty = 'easy',
  hideLogin,
}) => {
  if (freegame) {
    return (
      <Info
        id={1}
        disableCache
        name="freegameIps"
        path="/freegame_ips"
        method="POST"
        render={info => {
          if (info && info.ipBlocked) {
            navigate('/freegame-play-again')
          }

          return (
            <List
              disableCache
              name="products"
              path={`/products/${difficulty}`}
              render={ScreenWithSpeed(
                difficulty,
                navigate,
                currentUser,
                location,
                freegame,
                hideLogin,
              )}
            />
          )
        }}
      />
    )
  }

  return currentUser ? (
    <List
      name="products"
      path={`/products/${difficulty}`}
      render={ScreenWithSpeed(difficulty, navigate, currentUser, location)}
    />
  ) : (
    <Redirect to={`/sign-in?next=/game/${difficulty}`} noThrow />
  )
}
