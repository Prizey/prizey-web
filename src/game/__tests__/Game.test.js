import React from 'react'
import renderer from 'react-test-renderer'
import GameScreen from '../GameScreen'

const mockList = [
  { id: 1, image: '/mocks/trump-mask.png', title: 'trump' },
  { id: 2, image: '/mocks/sweatshirt.png', title: 'sweatshirt' },
  { id: 3, image: '/mocks/shoe.png', title: 'shoe' },
]

jest.mock('croods', () => ({
  Info: ({ children, ...props }) => (
    <div {...props}>
      Info -{' '}
      {props.render({
        easyCarouselSpeed: 500,
      })}
    </div>
  ),
  List: props => (
    <div {...props}>
      {props.children} - Render: {props.render(mockList)}
    </div>
  ),
  Provider: ({ children, ...props }) => (
    <div {...props}>Provider - {children}</div>
  ),
  createReducer: () => (state = {}) => state,
}))

it('renders correctly', () => {
  const tree = renderer.create(<GameScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('trigger the alert', () => {
  global.alert = jest.fn()
  const tree = renderer.create(<GameScreen />).root

  tree.findByProps({ speed: 500 }).props.onSelectItem({ title: 'item' })
  expect(global.alert).toHaveBeenCalledTimes(1)
})
