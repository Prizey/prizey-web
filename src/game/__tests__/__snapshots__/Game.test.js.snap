// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  name="products"
  path="/products/easy"
  render={[Function]}
>
   - Render: 
  <div
    name="game_settings"
    path="/game_settings"
    render={[Function]}
  >
     - Render: 
    <div
      className="Component-root-5"
    >
      <div
        className="Component-header-4"
      >
        <a
          aria-current="page"
          aria-label="Go Back"
          className="MuiButtonBase-root-13 MuiIconButton-root-7 Component-root-6"
          href="/"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiIconButton-label-12"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-16 MuiSvgIcon-fontSizeLarge-24"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-68"
          />
        </a>
      </div>
      <div
        className="Component-container-3"
      >
        <h2
          className="MuiTypography-root-25 MuiTypography-h2-38 MuiTypography-alignCenter-48"
        >
          PRIZEY
        </h2>
        <div
          className="Component-caption-1"
        >
          <div
            className="Component-root-65"
          >
            <span
              className="Component-icon-61 Component-icon_easy-62"
            />
            <h6
              className="MuiTypography-root-25 MuiTypography-subtitle1-43"
            >
              EASY
            </h6>
          </div>
        </div>
        <div
          className="Component-children-2"
        >
          <div
            className="Component-root-67"
          >
            <img
              alt="trump"
              className="Component-img-66"
              onClick={[Function]}
              pose="enter"
              src="/mocks/trump-mask.png"
            />
            <img
              alt="sweatshirt"
              className="Component-img-66"
              onClick={[Function]}
              pose="leave"
              src="/mocks/sweatshirt.png"
            />
            <img
              alt="shoe"
              className="Component-img-66"
              onClick={[Function]}
              pose="leave"
              src="/mocks/shoe.png"
            />
          </div>
        </div>
        <p
          className="MuiTypography-root-25 MuiTypography-body1-34 MuiTypography-alignCenter-48"
        />
      </div>
    </div>
  </div>
</div>
`;
